povm_toolbox/quantum_info/product_frame.py

```diff
@@ -201,6 +203,37 @@ def _check_validity(self) -> None:
         for povm in self._frames.values():
             povm._check_validity()
 
+    def _ravel_index(self, index: tuple[int, ...]) -> tuple[int, ...]:
+        """Process a global multi-index into a tuple of flat indices for each local frame.
+
+        Args:
+            index: a large multi-index consisting of local multi-indices, each of those
+                corresponding to a local frame.
```

Comment:
If this consists of multi-indices, does that not mean we can have `tuple[LabelMultiQubitT, ...]` as the input?
If so, that means you need to handle nested tuples below, too?

Reply:
The "large multi-index" is expected to be flat. That is, it is supposed to be a flattened `tuple[LabelMultiQubitT, ...]` which is therefore always a `tuple[int, ...]`.
I agree it wasn't clear/explained in the current docstring.. I will add this explanation to the docstring.

Reply:
Actually after looking again at the code I think I could adapt the structure such that `index` is a `tuple[LabelMultiQubitT, ...]`, it would make more sense and be more consistent.